site-src/concepts/api-overview.md

```diff
@@ -75,35 +76,42 @@ Status object.
 A Gateway MAY contain one or more *Route references which serve to direct
 traffic for a subset of traffic to a specific service.*
 
-### {HTTP,TCP,Foo}Route
+### Route objects
 
 Route objects define protocol-specific rules for mapping requests from a Gateway
 to Kubernetes Services.
 
-`HTTPRoute` and `TCPRoute` are currently the only defined Route objects.
-Additional protocol-specific Route objects may be added in the future.
+As of v1alpha2, there are four Route types defined in this repo, although it's
+expected that in the future implementations may create their own custom Route
+types.
 
-### BackendPolicy
+#### HTTPRoute
+HTTPRoute is for multiplexing HTTP or terminated HTTPS connections onto a single port or set of ports. It's intended for use in cases where you want to inspect the HTTP stream and use HTTP-level data for either routing or modification. (An example here is using HTTP Headers for routing, or modifying them in-flight).
 
-BackendPolicy provides a way to configure connections between a Gateway and a
-backend. For the purpose of this API, a backend is any resource that a route can
-forward traffic to. A common example of a backend is a Service. Configuration at
-this level is currently limited to TLS, but will expand in the future to support
-more advanced policies such as health checking.
+#### TLSRoute
+TLSRoute is for multiplexing TLS connections, discriminated via SNI, onto a single port or set or ports. It's intended for where you want to use the SNI as the main routing method, and are not interested in properties of the underlying connection (such as whether or not it is HTTP).
+
+#### TCPRoute and UDPRoute
+TCPRoute (and UDPRoute) are intended for use as a mapping between a single port or set of ports and a single backend. In this case, there is no discriminator you can use to choose different backends on the same port, so each TCPRoute really needs a different port on the listener (in general, anyway).
+
+#### Route summary table
+The "Routing Discriminator" column below refers to what information can be used to allow multiple Routes to share ports on the Listener.
+
+|Object|OSI Layer|Routing Discriminator|TLS Support|Purpose|
+|------|---------|---------------------|-----------|-------|
+|HTTPRoute| Layer 7 | Anything in the HTTP Protocol | Terminated only, can be reencrypted| HTTP and HTTPS Routing|
+|TLSRoute| Somewhere between layer 4 and 7| SNI or other TLS properties| Passthrough or terminated, can be reencrypted if terminated. | Routing of TLS protocols including HTTPS where inspection of the HTTP stream is not required.|
+|TCPRoute| Layer 4| None | None (but passthrough will work because the connection is forwarded) | Allows for forwarding of a TCP stream from the Listener to the Backends |
```

Comment:
TCPRoute can be TLS terminated or passed through, but not multiplexed using SNI. Is that distinction too complicated for this table?

Reply:
I don't think so, I'll fix that up. That's why I put the "Routing discriminator" column in.